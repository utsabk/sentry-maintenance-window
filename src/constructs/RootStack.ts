import { App, Stack, StackProps } from 'aws-cdk-lib';

export class RootStack extends Stack {
  constructor(scope: App, id: string, props: StackProps) {
    super(scope, id, props);

    console.log('hello');
  }
}
