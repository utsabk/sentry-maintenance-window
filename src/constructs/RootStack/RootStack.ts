import { App, Stack, StackProps } from 'aws-cdk-lib';

import { Scheduler } from '../Scheduler/Scheduler';

export class RootStack extends Stack {
  constructor(scope: App, id: string, props: StackProps) {
    super(scope, id, props);

    new Scheduler(this);
  }
}
