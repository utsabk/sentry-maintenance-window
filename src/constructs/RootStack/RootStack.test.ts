import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { RootStack } from './RootStack';

describe('RootStack', () => {
  it('produces a correct CloudFormation template', () => {
    const app = new App();
    const stack = new RootStack(app, 'test-app', {
      env: {
        account: 'dummy-account',
        region: 'dummy-region',
      },
    });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
