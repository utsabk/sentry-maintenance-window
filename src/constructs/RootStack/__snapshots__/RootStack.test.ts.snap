// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootStack produces a correct CloudFormation template 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "SchedulerNestedStackSchedulerNestedStackResourceA5A875E1": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.dummy-region.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/cdk-hnb659fds-assets-dummy-account-dummy-region/1b13607a75bafababdaee2c759a44c3f069c5d437f98dddc8fa4fa52e960e833.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
